master/buildbot/config.py

```diff
@@ -295,11 +295,16 @@ def copy_str_param(name, alt_key=None):
 
         copy_int_param('logCompressionLimit')
 
-        if 'logCompressionMethod' in config_dict:
-            logCompressionMethod = config_dict.get('logCompressionMethod')
-            if logCompressionMethod not in ('bz2', 'gz'):
-                error("c['logCompressionMethod'] must be 'bz2' or 'gz'")
-            self.logCompressionMethod = logCompressionMethod
+        self.logCompressionMethod = config_dict.get('logCompressionMethod', 'bz2')
+        if self.logCompressionMethod not in ('raw', 'bz2', 'gz', 'lz4'):
+            error("c['logCompressionMethod'] must be 'raw', 'bz2', 'gz' or 'lz4'")
+
+        if self.logCompressionMethod == "lz4":
+            try:
+                import lz4
+            except ImportError:
+                error("To set c['logCompressionMethod'] to 'lz4' you must install the lz4 library"
+                      "('pip install lz4'). Compression mode is set to 'raw'")
```

Comment:
Agreed, but the error message implies it is falling back to something else. Except it's an error so it can't fall back to anything.